src/browser/BrowserService.cpp

```diff
@@ -1053,7 +1053,7 @@ bool BrowserService::handleURL(const QString& entryUrl, const QString& url, cons
     }
 
     // Filter to match hostname in URL field
-    if (siteQUrl.host().endsWith(entryQUrl.host())) {
+    if (siteQUrl.host().endsWith(entryQUrl.host()) && baseDomain(siteQUrl.host()) == baseDomain(entryQUrl.host())) {
```

Comment:
Isn't this somewhat redundant?

Reply:
No. The subdomain matching is currently done with `endsWith` that can cause some extra matches appearing to the results. The baseDomain check ensusures the comparison is really made with the same host address. If you have a better proposition, I'll gladly take it.

Reply:
I mean, isn't the endsWith check redundant when you have the stricter baseDomain check in place?

Reply:
I find this comparison confusing because there is a lot of functions transforming the inputs to it. Needs to be simplified regardless or a comment on how it functions.

Reply:
Thanks for the feedback guys. I'll try to simplify the comparison.

Reply:
I made the baseDomain comparison as a separate if-sentence, so it's a little bit more clear.